src/main/kotlin/org/team5419/frc2019offseason/subsystems/Lift.kt

```diff
@@ -8,113 +8,154 @@ import com.ctre.phoenix.motorcontrol.InvertType
+            configPeakOutputForward(0.1)
+            configPeakOutputReverse(0.1)
+            enableCurrentLimit(false)
+            // configPeakCurrentDuration(0, 0)
+            // configPeakCurrentLimit(0, 0)
+            configContinuousCurrentLimit(25, 0) // amps
+            enableVoltageCompensation(false)
+            // configForwardSoftLimitThreshold(10000000, 0)
+            // configReverseSoftLimitThreshold(-1000000, 0)
+            // configForwardSoftLimitEnable(false, 0)
+            // configReverseSoftLimitEnable(false, 0)
         }
 
-        public override fun stop() {
-                setPosistion(LiftHeight.STOW)
-                mBrakeMode = false
+        mSlave = slaveTalon.apply {
+            follow(mMaster)
+            setInverted(InvertType.FollowMaster)
         }
 
-        public override fun reset() {
-                setPercent(0.0)
+        // setpoint = LiftHeight.getHeight()
+        setpoint = 0
+        firstStagePosistion = 0.0
+        secondStagePosistion = Lift.SECOND_STAGE_HIGHT
+        isSecondStage = false
+    }
+
+    public fun zero() {
+        setpoint = 0
+        firstStagePosistion = 0.0
+        secondStagePosistion = Lift.SECOND_STAGE_HIGHT
+        isSecondStage = false
+    }
+
+    public fun setPercent(speed: Double) {
+        mMaster.set(ControlMode.PercentOutput, speed)
+    }
+
+    public fun setPosistion(height: LiftHeight) {
+        if (firstStagePosistion < Lift.MAX_FLIP_HIGHT) {
+            setTicks(inchesToTicks(height.value))
         }
+    }
+
+    public fun setTicks(ticks: Int) {
+        println("posistion $ticks")
+        setpoint = ticks
+        mMaster.set(ControlMode.MotionMagic, ticks.toDouble())
+    }
+
+    public fun setInches(inches: Double) {
+        setTicks(inchesToTicks(inches))
+    }
+
+    public override fun update() {
+        // println(mMaster.getSelectedSensorPosition(0))
+        // if (!isSecondStage && firstStagePosistion + Lift.SECOND_STAGE_EPSILON > Lift.SECOND_STAGE_HIGHT) {
+        //     mMaster.apply {
+        //         config_kP(kElevatorSlot, Lift.KP2, 0)
+        //         config_kI(kElevatorSlot, Lift.KI2, 0)
+        //         config_kD(kElevatorSlot, Lift.KD2, 0)
+        //         config_kF(kElevatorSlot, Lift.KF2, 0)
+        //     }
+        //     isSecondStage = true
+        // } else if (isSecondStage && Lift.SECOND_STAGE_HIGHT + Lift.SECOND_STAGE_EPSILON > secondStagePosistion) {
+        //     mMaster.apply {
+        //         config_kP(kElevatorSlot, Lift.KP, 0)
+        //         config_kI(kElevatorSlot, Lift.KI, 0)
+        //         config_kD(kElevatorSlot, Lift.KD, 0)
+        //         config_kF(kElevatorSlot, Lift.KF, 0)
+        //     }
+        //     isSecondStage = false
+        // }
+        // println("Stage 1: " + firstStagePosistion.toString())
+    }
+
+    public override fun stop() {
+        setPosistion(LiftHeight.BOTTOM)
+        mBrakeMode = false
+    }

```

Comment:
the stop function should set the motors to 0.0 percent output, and set brake mode to true